app/src/main/java/acr/browser/lightning/settings/fragment/AdBlockSettingsFragment.kt

```diff
@@ -414,6 +415,38 @@ override fun onDestroy() {
         super.onActivityResult(requestCode, resultCode, data)
     }
 
+    // class necessary to allow separate onClickListener for the switch
+    private inner class FilterListSwitchPreference(context : Context?, val entity: AbpEntity) : SwitchPreferenceCompat(context) {
```

Comment:
Assuming this is something like Dark theme on LOS, could we generalized this to make it easily reusable? Name it SplitSwitchPreference or EditSwitchPreference I guess.

Reply:
It's a but different, and based on https://stackoverflow.com/questions/53159360/how-to-set-click-listener-to-this-button
Making it reusable should be possible, not sure how much work it would be.